target/linux/ath79/image/Makefile

```diff
@@ -81,9 +81,8 @@ endef
 
 define Device/loader-okli-uimage
   LOADER_TYPE := bin
-  COMPILE := loader-$(1).bin loader-$(1).uImage
-  COMPILE/loader-$(1).bin := loader-okli-compile
-  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+  COMPILE := loader-$(1).uImage
+  COMPILE/loader-$(1).uImage := loader-okli-compile | pad-to 64k | \
 	lzma | uImage lzma
```

Comment:
I'm a little curious why `pad-to 64k` here. It seems useless.

Reply:
that part is necessary, some bootloaders are not able to decompress it without the padding for some reason, maybe the uncompressed size affects the lzma stream properties somehow? i don't know...

anyway, padding is harmless and has negligible size difference in the final compressed file

maybe @xabolcs knows, he wanted to test different lengths of padding to see if there was a difference, but AFAIK there's no real difference

Reply:
@DragonBluep 

> I'm a little curious why `pad-to 64k` here. It seems useless.

Those ancient bootloaders doesn't properly flush dcache & invalidate icache before executing the code it just extracted. It wasn't a problem for linux kernel because they are relatively big and after completing the extraction, the beginning of the kernel code is already pushed out of the dcache. Meanwhile this LZMA loader is tiny, and the only way to get around this bug is to let the bootloader write a lot more empty data before executing.

Reply:
Okay, thanks for let me know that.😁